server/service/async/async.go

```diff
@@ -17,10 +14,11 @@ import (
 )
 
 const (
-	labelMembershipHostKeyPattern = "label_membership:{*}"
-	labelMembershipHostKey        = "label_membership:{%d}"
-	labelMembershipReportedKey    = "label_membership_reported:{%d}"
-	collectorLockKey              = "locks:async_collector:{%s}"
+	labelMembershipActiveHostIDsKey = "label_membership:active_host_ids"
+	labelMembershipHostKey          = "label_membership:{%d}"
+	labelMembershipReportedKey      = "label_membership_reported:{%d}"
+	labelMembershipKeysMinTTL       = 7 * 24 * time.Hour // 1 week
```

Comment:
Curious why so big TTL if the default `osquery.async_host_collect_interval` is `30s`

Reply:
It's just that there's not really a reason to be super aggressive on this, the goal is only that those dead hosts' keys will go away at some point, so I went with the same TTL we put on live queries (as minimum, and 10 times the collect interval if that is longer, e.g. if for some reason there's an incredibly long interval set for async collection, make sure the collector still gets a fair amount of runs to collect that data). Happy to tweak it to lower if folks feel strongly about it, though!